sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -569,16 +517,13 @@ export class EventHubReceiver extends LinkEntity {
         : eventsToRetrieveCount === 0
         ? Promise.resolve(this.queue.splice(0, maxMessageCount))
         : new Promise<void>((resolve, reject) => {
-            this._onError = (err) => {
-              if (err.name === "AbortError") {
-                cleanupBeforeAbort();
-              }
-              this.clearHandlers();
-              reject(err);
-            };
+            this._onError = reject;
```

Comment:
It is guaranteed that the connection will be closed when aborting because of the listeners attached on the abort events elsewhere, so no need to double perform that here again. Also, `clearHandlers()` will also be called `finally` here.